lib/ostruct.rb

```diff
@@ -88,12 +87,13 @@ class OpenStruct
   #
   #   data   # => #<OpenStruct country="Australia", capital="Canberra">
   #
-  def initialize(hash=nil)
+  def initialize(hash=nil, is_recursive=false)
```

Comment:
Would be more readable to use a keyword argument I'd say

Reply:
Updated! Thanks :)

Reply:
if we use keyword argument the below test will fail:

```
def test_accessor_defines_method
    os = OpenStruct.new(foo: 42)
    assert os.respond_to? :foo
    assert_equal([], os.singleton_methods)
    assert_equal(42, os.foo)
    assert_equal([:foo, :foo=], os.singleton_methods.sort)
end
```
But for better readability I changed the constructor to this:

```
def initialize(hash=nil, options={recursive: false})
    @table = {}
    @recursive = options.fetch(:recursive, false)
    if hash
      hash.each_pair do |k, v|
        k = k.to_sym
        @table[k] = (@recursive && v.respond_to?(:to_hash)) ? OpenStruct.new(v,true) : v
      end
    end
end
```